PPTT-4021 Updated guides to ecommerce v2 (#994)
PiwikPRO/PPMS-PublicDocs: data_collection/web/guides.rst

> E-commerce v2

Reviewer: I think we should call it just `E-commerce` without the `v2` mark.
In the guides, we want to inform customers about our suggested approach to implementing JSTC methods. We don't want to mention the old, deprecated way of tracking e-commerce here.

-------------

E-commerce v2 API supports 3 types of e-commerce interactions:

  * :ref:`Product views<guide_tracking_product_views>`
  * :ref:`Cart updates<guide_tracking_cart_updates>`
  * :ref:`Orders<guide_tracking_orders>`

.. _guide_tracking_product_views:

Tracking product views
^^^^^^^^^^^^^^^^^^^^^^

> Usually, the first action of a visitor related to e-commerce is checking products details. :ref:`ecommerceProductDetailView<jtc-api-ecommerceProductDetailView>` function allows us to track product view details (including product category).

Reviewer 2: ```suggestion
Usually, the first action of a visitor, related to e-commerce, is product details view. :ref:`ecommerceProductDetailView<jtc-api-ecommerceProductDetailView>` function allows us to track it (including product category).
```

Example:

.. code-block:: javascript

  _paq.push([
      "ecommerceProductDetailView",
      {
          sku: "craft-311",
          name: "Unicorn Iron on Patch",
          category: "Crafts & Sewing",
          price: "49.90",
          quantity: 3,
          brand: "DMZ",
          variant: "blue"
      }
  ]);

Attribute ``category`` of the product sent via :ref:`ecommerceProductDetailView<jtc-api-ecommerceProductDetailView>` function accepts not only string values, but also arrays of strings. This is useful for tracking products that belong to more than one category:

.. code-block:: javascript

  _paq.push([
      "ecommerceProductDetailView",
      {
          sku: "craft-311",
          name: "Unicorn Iron on Patch",
          category: ["Crafts & Sewing", "Toys"],
          price: "49.90",
          quantity: 3,
          brand: "DMZ",
          variant: "blue"
      }
  ]);

.. _guide_tracking_cart_updates:

Tracking cart updates
^^^^^^^^^^^^^^^^^^^^^

Another type of e-commerce activity you can track is an update of a shopping cart.

> With it, we are able to measure how often visitors add or remove specific products to the cart and what products stay in abandoned carts.

Reviewer 2: ```suggestion
It enables you to measure how often visitors add or remove specific products to the cart and what products stay in abandoned carts.
```

> Tracking a cart update has two steps:

Reviewer 2: Are those really `steps`? Wouldn't `modes` be a better word to describe this situation?

Author: I think it's steps because ideally you want to implement both. Modes for me implies that you can use one or the other but not both (usually).

  * cart update (:ref:`ecommerceCartUpdate<jtc-api-ecommerceCartUpdate>`)
  * product addition and removal (:ref:`ecommerceAddToCart<jtc-api-ecommerceAddToCart>` and :ref:`ecommerceRemoveFromCart<jtc-api-ecommerceRemoveFromCart>`)

Cart update allows us to synchronize longer living cart session (containing previously selected products) with short living visitor session.

> This command is optional but recommended. It should be used once per page (immidiately after loading).

Reviewer: `immediately` instead of `immidiately`

Reviewer: I think we should also add some articles here.
`A cart update allows us to synchronize a longer living cart session [...] with a shorter living visitor session.`

Example:

.. code-block:: javascript

  _paq.push([
      "ecommerceCartUpdate",
      [
          {
              sku: "craft-311",
              name: "Unicorn Iron on Patch",
              category: "Crafts & Sewing",
              price: "50.00",
              quantity: 3,
              brand: "DMZ",
              variant: "blue"
          }
          {
              sku: "craft-312",
              name: "Unicorn Iron on Grass",
              category: "Crafts & Sewing",
              price: "30.00",
              quantity: 1,
              brand: "DMZ",
              variant: "red"
          }
      ],
      "180.00"
  ]);

> Product addition and removal commands are used immidiately when visitor adds to a cart or removes from a cart any product.

Reviewer: Same as above, `immediately`. Also, remember about "a/an" articles - it should be `when a visitor` not `when visitor`.

Reviewer 2: ```suggestion
Product addition and removal commands are used immediately when a visitor adds to a cart or removes from a cart any product.
```

> These commands let us track how visitor interacted with a cart and update a cart state.

Reviewer: Correction: `how a visitor` instead of `how visitor`.

> Adding prodict to a cart:

Reviewer: `Adding a product to a cart`.

.. code-block:: javascript

  _paq.push([
      "ecommerceAddToCart",
      {
          sku: "craft-311",
          name: "Unicorn Iron on Patch",
          category: "Crafts & Sewing",
          price: "49.90",
          quantity: 3,
          brand: "DMZ",
          variant: "blue"
      }
  ]);

Removing product from a cart:

.. code-block:: javascript

  _paq.push([
      "ecommerceRemoveFromCart",
      {
          sku: "craft-311",
          name: "Unicorn Iron on Patch",
          category: "Crafts & Sewing",
          price: "49.90",
          quantity: 3,
          brand: "DMZ",
          variant: "blue"
      }
  ]);

.. _guide_tracking_orders:

Tracking orders
^^^^^^^^^^^^^^^

Perhaps the most important element of an e-commerce implementation is tracking orders.

> E-commerce v2 API is using :ref:`ecommerceConversion<jtc-api-ecommerceConversion>` for this task.

Reviewer 2: ```suggestion
E-commerce API offers :ref:`ecommerceConversion<jtc-api-ecommerceConversion>` for this task.
```
In the first comment @harmasz suggested no to call it `v2` - thus my code suggestion here.
